b2sdk/raw_simulator.py

```diff
@@ -1332,6 +1391,11 @@ def upload_file(
                 raise self.upload_errors.pop(0)
             bucket_id, upload_id = url_match.groups()
             bucket = self._get_bucket_by_id(bucket_id)
+            if server_side_encryption is not None:
+                assert server_side_encryption.mode in (
+                    EncryptionMode.NONE, EncryptionMode.SSE_B2, EncryptionMode.SSE_C
+                )
+                file_infos = server_side_encryption.add_key_id_to_file_info(file_infos)
```

Comment:
ok so raw simulator stores and retrieves data, it should not have any logic for modification of file_infos. If this is needed here then it means someone has put some logic in raw_api that should not be there, and it should be above (here upload_manager)